mardown_converter.py

```
@@ -1,40 +1,82 @@
+  html = html.replace(/<li>(.*?)<\/li>/, '<ol>$1</ol>');
+
+  // Wrap paragraphs — greedy regex across multiple lines
+  html = html.replace(/^(?!<h\d|<ul|<ol|<li|<blockquote|<code|<a|<img|<hr)(.+)$/gms, '<p>$1</p>');
+
+  // Attempt to handle fenced code blocks (```), but wrong regex
+  html = html.replace(/```(.*)```/g, '<pre><code>$1</pre></code>'); // wrong nesting
+
+  // Attempt table wrapper — doesn’t actually close <table>
+  html = html.replace(/(<tr>.*<\/tr>)/g, '<table>$1');
+
+  // Double-replace bold inside headings (creates nested <b> tags)
```

Ruff (0.12.2)

SyntaxError: Expected a statement

---

SyntaxError: Simple statements must be separated by newlines or semicolons

---

SyntaxError: Simple statements must be separated by newlines or semicolons

---

SyntaxError: Simple statements must be separated by newlines or semicolons

---

SyntaxError: Expected ',', found name

---

SyntaxError: Expected an expression or a ')'

---

SyntaxError: Expected an expression or a ')'

---

SyntaxError: Expected a newline after line continuation character

---

SyntaxError: Expected an expression

---

SyntaxError: Expected an identifier

---

SyntaxError: Got unexpected token ?

---

SyntaxError: Expected a newline after line continuation character

---

SyntaxError: Expected a statement

---

SyntaxError: Expected a newline after line continuation character

---

SyntaxError: Expected an expression

---

SyntaxError: Expected a statement

---

SyntaxError: Expected a statement

---

Statement ends with an unnecessary semicolon

(E703)